### PR TITLE
init js sharing infrastructure

### PR DESCRIPTION
**Pull request by wibjorn**
Task: task-[work-item-number]

Link: preview-374

The publishing mechanism can only _really_ be verified after this PR has merged. However, we can also double check that change haven't affected anything negatively in the site.

## Testing

1. Visit site and popover page. 
2. Verify JS still working fine.
3. Run the following command and inspect generated files in js/dist

```
npm run build:js
```

### CODE REVIEW

File: js/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+- A component's markup / attributes require changing upon interaction for compatibilty with screenreaders or other assistive technology.
+- A particular pattern is a good candidate for a re-usable web component. That is, its behavior is relatively straightforward and lightweight.
+
+## Installing this package
+
+Install the `@microsoft/atlas-js` package in your project's dependencies.
+
+```
+npm i @microsoft/atlas-js;
```
Comment (BenDMyers) [inline js/README.md:13]: nit: Does this need the semicolon?

File: js/README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+```
+
+## Accesssing its Typescript through the `src` folder
```
Comment (BenDMyers) [inline js/README.md:16]: nit: There's a third, extra `s` in `Accesssing` here

**Review by BenDMyers — CHANGES_REQUESTED** [review]

File: js/README.md [hunk 3]
```diff
@@ -1,3 +1,34 @@
-## Atlas JS
+# Atlas JS
 
-TBD.
+Welcome to TypeScript/JavaScript code backing the Atlas Design system. While Atlas endeavors to find CSS-only strategies for creating modern UI patterns, there are times when writing behavior code is necessary. These cases tend to fall into a one of several categories.
+
+- A component's markup / attributes require changing upon interaction for compatibilty with screenreaders or other assistive technology.
```
Comment (homing1) [inline js/README.md:5]: "compatibility"

Comment (homing1) [inline js/README.md:5]: "compatibility"

File: js/esbuild.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
+const argv = require('yargs').argv;
+const { watch } = argv;
```
Comment (jdanyow) [inline js/esbuild.js:2]: yargs is a little heavy for what we need here- could use `const watch = process.argv.includes('--watch');`

File: js/esbuild.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+const esbuild = require('esbuild');
+
+const target = 'es2020';
+const outdir = 'dist';
+const sourcedir = 'src';
+// This message must match the script ready regexp in packages\devops\tasks\develop.js
+const successMessage = '[esbuild] scripts build succeeded';
```
Comment (jdanyow) [inline js/esbuild.js:9]: I don't think outdir, sourcedir, successMessage are needed in this version

File: js/esbuild.js [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+			entryPoints: ['./src/index.ts'],
+			outdir,
+			bundle: true,
+			sourcemap: false,
+			minify: true,
+			treeShaking: true,
+			target
+		});
+		console.log(successMessage);
+	} catch (e) {
```
Comment (jdanyow) [inline js/esbuild.js:29]: no need for the try-catch here

File: js/README.md (hunk 3 above)
Comment (jdanyow) [inline js/README.md:5]: thinking the folders should be something like this
```
behaviors/           // behaviors added by decorating an element with classes or data attributes
  dismiss.ts
  popover.ts
  show-more.ts
elements/           // web components
  local-time.ts
  form-validation.ts

Comment (wibjorn) [reply js/README.md:5]: I like it

File: js/README.md (hunk 1 above)
Comment (wibjorn) [reply js/README.md:13]: nope!

File: js/README.md (hunk 3 above)
Comment (wibjorn) [reply js/README.md:5]: excellent, updated

Comment (wibjorn) [reply js/README.md:5]: added space, fixed compat

File: js/esbuild.js (hunk 4 above)
Comment (wibjorn) [reply js/esbuild.js]: updated

File: js/esbuild.js (hunk 5 above)
Comment (wibjorn) [reply js/esbuild.js]: updated

File: js/esbuild.js (hunk 6 above)
Comment (wibjorn) [reply js/esbuild.js]: updated

File: js/README.md (hunk 3 above)
Comment (wibjorn) [reply js/README.md]: updated

**Review by ollips — CHANGES_REQUESTED** [review]
I began reviewing the [popover page](https://design.docs.microsoft.com/pulls/374/components/popover.html) and seems like none of the popover's events are working 🤔 I'm able to open several popovers at the same time, clicking on `esc` button doesn't close the popover, etc.

**Review by ollips — COMMENTED** [review]
Duplicating the comment to keep the discussion in one place

I began reviewing the [popover page](https://design.docs.microsoft.com/pulls/374/components/popover.html) and seems like none of the popover's events are working 🤔 I'm able to open several popovers at the same time, clicking on esc button doesn't close the popover, etc.

**Review by jdanyow — APPROVED** [review]

**Review by ollips — APPROVED** [review]

**Pull request merged**